backend/modules/browser/utils.py

```diff
@@ -361,33 +363,34 @@ def get_variant_list(dataset:str, datatype:str, item:str, ds_version:str=None):
             return None
         refgene = transcript['gene_id']
 
-    for variant in variants:
-        if datatype in ('gene', 'transcript'):
-            anno = None
-            if datatype == 'transcript':
-                anno = [ann for ann in variant['vep_annotations'] if ann['Feature'] == item]
-                if not anno:
-                    anno = [ann for ann in variant['vep_annotations'] if ann['Gene'] == refgene]
-            else:
-                anno = [ann for ann in variant['vep_annotations'] if ann['Gene'] == item]
-            if anno:
-                variant['vep_annotations'] = anno
-
-    add_consequence_to_variants(variants)
-
-    for variant in variants:
-        remove_extraneous_information(variant)
-
-    # Format output
-    def format_variant(variant):
-        variant['major_consequence'] = (variant['major_consequence'].replace('_variant','')
-                                        .replace('_prime_', '\'')
-                                        .replace('_', ' '))
-
-        # This is so an array values turns into a comma separated string instead
-        return {k: ", ".join(v) if isinstance(v,list) else v for k, v in variant.items()}
-
-    variants = list(map(format_variant, variants))
+    if variants:
+        for variant in variants:
+            if datatype in ('gene', 'transcript'):
```

Comment:
I know this is not what was focused on in the PR, but this `if` statement seems superflous to me. Wouldn't it be better with this approximate structure:

```python
anno = None
if datatype == 'transcript':
    # ...
elif datatype == 'gene':
    # ...
if anno:
    variant['vep_annotations'] = anno
```

This code is too complex in general, it needs to be simplified somehow but I don't know quite how at the moment. Maybe a solution is to have different classes for gene, region and transcript and dispatch based on that. Not now though.